src/Edge.cpp

```diff
@@ -1,6 +1,8 @@
 #include "Edge.h"
 #include <iostream>
 
+Edge::Edge() {}
```

Comment:
why do you need this?  this should be unnecessary because this is a default constructor.